projects/storefrontlib/shared/components/generic-link/generic-link.component.ts

```diff
@@ -77,6 +77,18 @@ export class GenericLinkComponent implements OnChanges {
     return this.routeParts.path;
   }
 
+  get linkUrl(): string {
+    let url: string;
+    if (this.isExternalUrl()) {
+      url = this.url as string;
+    } else {
+      const homeUrlTree = this.router.createUrlTree(['']);
+      const homeUrl = this.router.serializeUrl(homeUrlTree).slice(0, -1);
+      url = `${homeUrl}${this.url}`;
```

Comment:
1. will it work correctly also when `this.url` contains site context? e.g. `this.url` is `/electronics-spa/en/USD/faq`. I guess not - I believe it would return `<site.com>/electronics-spa/en/USD/electronics-spa/en/USD/faq` (doubled site context)

2. did you consider instead using the: `this.routingService.getFullUrl(this.url)`?